core/tee/tee_svc_cryp.c

```diff
@@ -476,7 +476,7 @@ const struct tee_cryp_obj_type_attrs tee_cryp_obj_ed25519_keypair_attrs[] = {
 
 	{
 	.attr_id = TEE_ATTR_ECC_CURVE,
-	.flags = TEE_TYPE_ATTR_SIZE_INDICATOR | TEE_TYPE_ATTR_GEN_KEY_REQ,
+	.flags = TEE_TYPE_ATTR_SIZE_INDICATOR | TEE_TYPE_ATTR_GEN_KEY_OPT,
```

Comment:
Is it really what we want? See https://github.com/OP-TEE/optee_os/pull/5573. According to the GP spec (1.3.1) `TEE_ATTR_ECC_CURVE` is NOT a valid attribute for `TEE_TYPE_ED25519_KEYPAIR`.

Reply:
Agree, there no need for the CURVE attribute.
Previously the key size was deduced based on curve when
in CMD_CREATE_OBJECT.
It is reworked to get key size from PKCS11_CKA_EC_POINT.
How do you think, is it Ok?
Also removed the code for ECC_CURVE attribute.